### PR TITLE
People/render people

### PR DESCRIPTION
**Pull request by m4dzix**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -16,26 +28,28 @@ export default () => (
             <Title>Movies Browser</Title>
           </LogoContainer>
           <LinkContainer>
-            <ListItem><StyledNavLink to="/popular-movies">movies</StyledNavLink></ListItem>
-            <ListItem><StyledNavLink to="/popular-people">people</StyledNavLink></ListItem>
+            <ListItem>
```
Comment (KacperSzemraj) [inline src/App.js:31]: Great, it looks much better now!

File: src/App.js [hunk 2]
```diff
@@ -16,26 +28,28 @@ export default () => (
...
+              <StyledNavLink to="/popular-movies">movies</StyledNavLink>
+            </ListItem>
+            <ListItem>
+              <StyledNavLink to="/popular-people">people</StyledNavLink>
+            </ListItem>
```
Comment (KacperSzemraj) [inline src/App.js:36]: Great, it looks much better now!

File: src/App.js [hunk 3]
```diff
@@ -2,9 +2,21 @@ import React from "react";
...
+  NavContainer,
+  NavContent,
+  Input,
+  InputContainer,
+  LogoContainer,
+  Title,
+  LinkContainer,
+  StyledImg,
+  ListItem,
+} from "./styled";
```
Comment (KacperSzemraj) [inline src/App.js:16]: This is much more readable ;)

File: src/common/Section/style.js [hunk 4]
```diff
@@ -1,24 +1,42 @@
...
     grid-template-columns: repeat(2, 324px);
   }
+
   @media (max-width: ${({ theme }) => theme.breakpoints.mobileMax}px) {
     grid-template-columns: 1fr;
   }
+
+  ${(props) =>
+    props.people &&
+    css`
```
Comment (KacperSzemraj) [inline src/common/Section/style.js:20]: Wouldn't it be better to write in one line?

File: src/common/Section/style.js [hunk 5]
```diff
@@ -1,24 +1,42 @@
...
+      grid-template-columns: repeat(6, 208px);
+
+      @media (max-width: ${({ theme }) => theme.breakpoints.tablet}px) {
+        grid-template-columns: repeat(4, auto);
+      }
```
Comment (KacperSzemraj) [inline src/common/Section/style.js:25]: Do you think there is no need to add a grid-gap for tablets?

File: src/common/Tile/index.js [hunk 6]
```diff
@@ -11,22 +11,30 @@ import {
...
+          <Title people={group}>{title}</Title>
           <Year>{year}</Year>
           <Tags>{type}</Tags>
         </DetailsContainer>
         <VoteContainer>
-          <img src={starIcon} alt={"star"}></img>
+          <img src={starIcon} alt={""}></img>
           <VoteAverage>{voteAverage}</VoteAverage>
-          <VoteCount>{voteCount} votes</VoteCount>
+          <VoteCount>{voteCount} </VoteCount>
```
Comment (KacperSzemraj) [inline src/common/Tile/index.js]: Compare yourself above, is the space needed after closing curly brace?

File: src/common/Tile/style.js [hunk 7]
```diff
@@ -17,43 +17,87 @@ export const Container = styled.div`
     height: auto;
     grid-template-columns: auto 1fr;
   }
+
+  ${(props) =>
+    props.people &&
+    css`
```
Comment (KacperSzemraj) [inline src/common/Tile/style.js:23]: As before, it can be in one line. Why this practice?

File: src/store.js [hunk 8]
```diff
@@ -1,11 +1,13 @@
 import { configureStore } from "@reduxjs/toolkit";
 import createSagaMiddleware from "redux-saga";
 import moviesReducer from "./features/PopularMovies/moviesSlice";
+import peopleReducer from "./features/People/peopleSlice";
 import rootSaga from "./rootSaga";
 
 const sagaMiddleware = createSagaMiddleware();
 const store = configureStore({
-  reducer: { movies: moviesReducer },
+  reducer: { movies: moviesReducer, people: peopleReducer },
```
Comment (KacperSzemraj) [inline src/store.js:9]: OK, looks fine!

File: src/features/PopularMovies/index.js [hunk 9]
```diff
@@ -12,6 +12,7 @@ import {
   selectGenres,
 } from "./moviesSlice";
 import { useSelector, useDispatch } from "react-redux";
+import starIcon from "../../assets/Vector.svg";
```
Comment (KacperSzemraj) [inline src/features/PopularMovies/index.js:15]: Don't you think you can call this icon a little different than Vector.svg? Maybe more specifically. Of course it's a trifle, but for the sake of consistency I think that name star-ico.svg will be ok. ;)

File: src/features/People/peopleSlice.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+});
+export const {
+  fetchPopularPeople,
+  fetchPopularPeopleSuccess,
+  fetchPopularPeopleError,
+} = peopleSlice.actions;
+export const selectPeopleState = (state) => state.people;
+export const selectPopularPeople = (state) => selectPeopleState(state).people;
+export const selectGroup = (state) => selectPeopleState(state).group;
+export const selectLoading = (state) => selectPeopleState(state).loading;
```
Comment (KacperSzemraj) [inline src/features/People/peopleSlice.js]: Here above you can make a distance to make the selectors more readable. ;) Besides, it looks great.

File: src/features/People/peopleSlice.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    fetchPopularPeople: (state) => {
+      state.loading = true;
+    },
+    fetchPopularPeopleSuccess: (state, { payload: people }) => {
+      state.people = people.results;
+      state.group = true;
+      state.loading = false;
+    },
+    fetchPopularPeopleError: (state) => {
+      state.loading = false;
```
Comment (KacperSzemraj) [inline src/features/People/peopleSlice.js]: Very good reducers ;)

File: src/features/People/peopleSaga.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    const people = yield call(getPopularPeople);
+    yield put(fetchPopularPeopleSuccess(people));
+  } catch (error) {
+    yield put(fetchPopularPeopleError());
+  }
+}
+
+export function* watchFetchPopularPeople() {
+  yield takeEvery(fetchPopularPeople.type, fetchPopularPeopleHandler);
+}
```
Comment (KacperSzemraj) [inline src/features/People/peopleSaga.js:21]: OK, great!

File: src/features/People/PopularPeople/index.js [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+        <Section
+          group={group}
+          title={"Popular people"}
+          body={popularPeople.map((people) => (
+            <Tile
+              group={group}
+              key={people.id}
+              title={people.name}
+              imagePath={`https://image.tmdb.org/t/p/w185/${people.profile_path}`}
+            ></Tile>
```
Comment (KacperSzemraj) [inline src/features/People/PopularPeople/index.js:38]: I'm not sure, but in such places, I'd leave the angle bracket separately which closes the component. What do you think about it?

File: src/features/People/getPeopleApi.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  const response = await fetch(api);
+
+  if (!response.ok) {
+    throw new Error(response.statusText);
+  }
+
+  const popularPeople = await response.json();
+
+  return popularPeople;
+};
```
Comment (KacperSzemraj) [inline src/features/People/getPeopleApi.js:14]: Ok, it will work ;)

**Review by KacperSzemraj — APPROVED** [review]
Hello, 

I checked your changes, generally you can merge. Check the formatting only in certain places. Of course, later on we will determine together how and why we will format the code in the project. Everyone has different practices. ;)

Regards, Kacper

File: src/common/Section/style.js (hunk 4 above)
Comment (m4dzix) [reply src/common/Section/style.js:20]: Maybe, but I have "Prettier" installed - it formats the code this way, so I didn't change it 🙂

File: src/common/Section/style.js (hunk 5 above)
Comment (m4dzix) [reply src/common/Section/style.js:25]: There is grid-gap - 24px. I just don't change it

File: src/common/Tile/index.js (hunk 6 above)
Comment (m4dzix) [reply src/common/Tile/index.js]: I didn't notice it 😆  . I'm already correcting

File: src/common/Tile/style.js (hunk 7 above)
Comment (m4dzix) [reply src/common/Tile/style.js:23]: Prettier again 😉

File: src/features/People/PopularPeople/index.js (hunk 13 above)
Comment (m4dzix) [reply src/features/People/PopularPeople/index.js:38]: The same - Prettier 😄

File: src/features/People/peopleSlice.js (hunk 10 above)
Comment (m4dzix) [reply src/features/People/peopleSlice.js]: Good advice. I'll add some space

File: src/features/People/peopleSlice.js (hunk 11 above)
Comment (m4dzix) [reply src/features/People/peopleSlice.js]: Thanks 😃

**Pull request merged**